core/chains/evm/log/eth_subscriber.go

```diff
@@ -255,6 +255,7 @@ func (sub managedSubscriptionImpl) Logs() chan types.Log {
 
 func (sub managedSubscriptionImpl) Unsubscribe() {
 	sub.subscription.Unsubscribe()
+	<-sub.Err() // ensure sending has stopped before closing the chan
```

Comment:
This line is the crux of the fix. Everything else is supporting this or directly related cascading changes/simplifications.